packages/gatsby/src/internal-plugins/internal-data-bridge/gatsby-node.js

```diff
@@ -87,8 +85,8 @@ exports.sourceNodes = ({ createContentDigest, actions, store }) => {
       siteMetadata: {
         ...configCopy.siteMetadata,
       },
-      port: state.program.proxyPort,
-      host: state.program.host,
```

Comment:
Here we retain the state

Reply:
Yup, we could call `store.getState()` here, but because `program` can't change this is fine (`program` does get retained in memory either way)